solver-wrapper.js

```diff
@@ -0,0 +1,244 @@
+		for(var index in expressions){
+			var expression = expressions[index];
+			var eq = {};
+			if(typeof(expression) == "string" && expression.indexOf("=") > -1){
+				var eqs = expression.split("=");
+				if(eqs.length > 2)
+					throw {
+						type: "equation.incorrect.format",
+						message: messages.get("equation.incorrect.format")
+					};
+				eq.lhs = Parser.parse(eqs[0]);
+				eq.rhs = Parser.parse(eqs[1]);
+			} else if(typeof(expression) == "string"){
+				eq.lhs = Parser.parse(expression);
+				eq.rhs = Parser.parse(0);
+			} else if(typeof(expression) == "object" && expression.length == 2){ 
+				eq.lhs = expression[0];
+				eq.rhs = expression[1];
+			} else if(typeof(expression) == "object"){
+				eq.lhs = expression;
+				eq.rhs = Parser.parse(0);
+			}
+			if(_isValid(eq)){
+				equations.push(eq);
+				_addVariables(eq);
+			}
+		}
+
+		if(variables.length == 0)
+			throw {
+				type: "no.variables",
+				message: messages.get("no.variables")
+			};
+	}
+
+	/**
+	* used to add unknown variables in the variables array created at the top.
+	* @param -  expression - from which unknown variables are to be added
+	**/
+	var _addVariables = function(/* object */ expression){
+		var update = function(variableArray){
+			for(var index in variableArray){
+				var variable = variableArray[index];
+				if(variables.indexOf(variable) < 0)
+					variables.push(variable);
+			}
+		};
+		update(expression.lhs.variables());
+		if(expression.rhs) update(expression.rhs.variables());
+	};
+
+	/**
+	* Preprocessing of equations will be part of this function. Currently preprocessing
+	* involves removing that equation which does not have any variable in it. 2 = 2 sort
+	* of equations.
+	* @param -  expression - equation which is to checked
+	* @return - isValid - boolean to provide whether expression is correct or not.
+	**/
+	var _isValid = function(/* object */ expression){
+		var isValid = true;
+		if((!expression.lhs || expression.lhs.variables().length == 0) &&
+			(!expression.rhs || expression.rhs.variables().length == 0))
+			isValid = false;
+
+		return isValid;
+	};
+
+	/**
+	* for over determined system where number of equations are more than the number
+	* of variables, this picks up a subset based on removing equations which do not
+	* bring any new variables in the system. First they are labelled as redundant
+	* and then the appropriate equations are removed.
+	* @param -	expressions - array for system of equations to be solved
+	*        	start - this is a search algorithm, index is the starting equation
+	* @return -	excess - the index of equations that were removed from expressions array
+	**/
+	var _preprocess = function(/* Array */ expressions, /* integer */ start){
+		var l = expressions.length;
+		var excess = [];
+		var checked = []; // ids that are part of the system
+		var equationVariables = [];
+		var _comparator = function(a, b){
+			return a - b;
+		};
+		var index = start;
+		
+		for(var i in expressions){
```

Comment:
Using for...in to iterate over indices of an array is discouraged because for...in is made for objects, not arrays specifically (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Description).  They suggest switching to `for(var i = 0; i < expressions.length; i++)`  for iterating over indices.

Reply:
Done